src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs

```diff
@@ -113,27 +116,32 @@ internal string FormatScriptCommand(string scriptCommandline, string scriptExten
             return command;
         }
 
-        internal string ReplaceScriptVariables(string command)
+        internal string ReplaceScriptVariables(string scriptCommandline)
         {
-            foreach (var scriptVariableEntry in ScriptVariableToMSBuildMap)
+            Func<string, string> scriptVariableReplacementDelegate = key =>
             {
-                var scriptVariableName = scriptVariableEntry.Key;
-                var msbuildMapping = scriptVariableEntry.Value;
-
-                if (command.Contains($"%{scriptVariableName}%"))
+                if (ScriptVariableToMSBuildMap.ContainsKey(key))
```

Comment:
If the key is not in the map, it returns the key wrapped as an msbuild variable, that name change seems reasonable to me.